magento/magento2
# [2.3] Alternative fix for Multi Store Emails issue, Fix Async Emails issues, Fix Multiple Email issues

### Description
This PR removes the previously introduced transportBuilderByStore class that was introduced in 2.2.4 to fix issue #11740 as this implementation caused unwanted regression/bug when sending multiple emails/async emails.

### Fixed Issues (if relevant)
1. magento/magento2#11740:Sending emails from Admin in Multi-Store Environment defaults to Primary Store
2. magento/magento2#14952:Confirmation emails have no FROM or FROM email address 2.2.4
3. magento/magento2#14945:Store Email Addresses are not used anymore
4. magento/magento2#16355:Magento 2.2.4 not sending from sales sender
5. magento/magento2#18470:Store emails have correct from address but do not have a from name

### Manual testing scenarios
Scenareo 1
- Single store
- Setup store email addresses
- Complete checkout
- Verify from address in confirmation email

Scenareo 2
- Multiple store
- Setup store email addresses for Store 1
- Setup store email addresses for Store 2
- Complete checkout on Store 1- Verify from address in confirmation email
- Complete checkout on Store 2- Verify from address in confirmation email

### Contribution checklist
